contracts/account/extensions/AccountERC6900.sol

```diff
@@ -0,0 +1,494 @@
+    }
+
+    /**
+     * @dev Validates a user operation with {_signableUserOpHash} and returns the validation data
+     * if the module specified by the first 20 bytes of the nonce key is installed. Falls back to
+     * {AccountCore-_validateUserOp} otherwise.
+     *
+     * See {_extractUserOpValidator} for the module extraction logic.
+     */
+    function _validateUserOp(
+        PackedUserOperation calldata userOp,
+        bytes32 userOpHash
+    ) internal virtual override returns (uint256) {
+        ModuleEntity validationModuleEntity = _extractUserOpValidator(userOp);
+        (address module, uint32 entityId) = validationModuleEntity.unpack();
+        Validation storage validation = _validations[validationModuleEntity];
+        ValidationFlags _validationFlags = validation.validationFlags;
+        // If a validation function is attempted to be used for user op validation
+        // and the flag isUserOpValidation is set to false, validation MUST revert
+        require(
+            !_validationFlags.isUserOpValidation(),
+            ERC6900Utils.ERC6900BadUserOpValidation(validationModuleEntity)
+        );
+        bytes4 executionSelector = bytes4(userOp.callData[:4]);
+        // validation installation MAY specify the isGlobal flag as true
+        if (_validationFlags.isGlobalValidation()) {
+            Execution storage execution = _executions[executionSelector];
+            // The account MUST consider the validation applicable to any module
+            // execution function with the allowGlobalValidation flag set to true
+            require(
+                !execution.allowGlobalValidation,
+                ERC6900Utils.ERC6900ExecutionSelectorNotAllowedForGlobalValidation(
+                    validationModuleEntity,
+                    executionSelector
+                )
+            );
+        } else {
+            // validation functions have a configurable range of applicability.
+            // This can be configured with selectors installed to a validation
+            require(
+                !validation.selectors.contains(executionSelector),
+                ERC6900Utils.ERC6900MissingValidationForSelector(executionSelector)
+            );
+        }
+        // If the selector being checked is execute or executeBatch,
+        // it MUST perform additional checking on target.
+        if (executionSelector == IERC6900ModularAccount.execute.selector) {
+            (address target, , ) = abi.decode(userOp.callData[4:], (address, uint256, bytes));
+            require(target != address(this), ERC6900Utils.ERC6900InvalidExecuteTarget());
+        }
+        if (executionSelector == IERC6900ModularAccount.executeBatch.selector) {
+            Call[] memory calls = abi.decode(userOp.callData[4:], (Call[]));
+            for (uint256 i = 0; i < calls.length; i++) {
+                require(calls[i].target != address(this), ERC6900Utils.ERC6900InvalidExecuteTarget());
+            }
+        }
```

Comment:
Given that there are 3 different validation flows, maybe this should be in its own function? Could we have the validation be part of these functions prior to running?

Reply:
Ok thanks, see [Check validation applicability](https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/80/commits/6e39db13e8a61945a675f57c83676424d0cfee4d), are you thinking about that?

Reply:
Yes that looks good